gazebo_ros/include/gazebo_ros/gazebo_ros_api_plugin.h

```diff
@@ -380,6 +385,12 @@ class GazeboRosApiPlugin : public SystemPlugin
   int pub_clock_frequency_;
   gazebo::common::Time last_pub_clock_time_;
 
+  /// \brief Sim time buffer for real-time factor calculation
+  std::list<common::Time> simTimes;
+
+  /// \brief Real time buffer for real-time factor calculation
+  std::list<common::Time> realTimes;
```

Comment:
[boost::circular_buffer](https://www.boost.org/doc/libs/1_65_1/doc/html/circular_buffer.html) would save some hassle with these

Reply:
I think that's a good idea.

Reply:
We've been trying to reduce our boost dependencies, especially when it's straight forward to implement the same thing using the standard library, like in this case.

Here, since Gazebo and gazebo_ros_pkgs already bring Boost as a dependency, this code is not adding a new dependency, so I think it's fine to leave it since an effort has been made to port it to use boost. But beware that we will most likely need to do the opposite effort of removing boost from here in the not-so-near future.